06-walkthrough-1.Rmd

```diff
@@ -133,20 +142,34 @@ reverse_scale <- function(question) {
   x
 }
 
+# Reverse scale for questions 4 and 7
 s12_pre_survey <- s12_pre_survey %>%
   mutate(q4 = reverse_scale(q4),
          q7 = reverse_scale(q7))
 
-# Int: 1, 4, 5, 8, 10
-# UV: 2, 6, 9
-# PC: 3, 7
-
-# need to change these to be within a mutate
-s12_pre_survey <- s12_pre_survey %>%
-  mutate(int = mean(q1, q4, q8, q10, q5),
-         uv = mean(q2, q6, q9),
-         percomp = mean(q3, q7),
-         tv = mean(q1, q2, q5, q6, q8, q9, q10))
+# Add measure variable 
+s12_measure_mean <- s12_pre_survey %>% 
+    # Gather questions and responses 
+    gather(question, response, c(q1:q10)) %>% 
+    mutate(
+        # Initialize measure variable as character
+        measure = "", 
+        # Interest measure
+        measure = ifelse(question %in% c("q1", "q4", "q5", "q8", "q10"), "int", measure), 
+        # Utility value measure
+        measure = ifelse(question %in% c("q2", "q6", "q9"), "uv", measure), 
+        # Perceived competence measure
+        measure = ifelse(question %in% c("q3", "q7"), "pc", measure)
```

Comment:
although this is fine, my tendency is to use `case_when` once, which makes it less likely you end up with `""`. If not, I would initialize measure with `NA_character_`, not a blank character.

Reply:
Oh, let me try `case_when`! For some reason I thought `case_when` needed a mathematical computation on the LHS of the formula as opposed to a logical with character classes. Standby!

Reply:
`case_when` just needs `predicate ~ value`, with two additions -- all rows must evaluate to `TRUE` in at least one predicate and `value` all have to be of the same type. That's why you often see `TRUE` on the LHS at the end (to ensure any non-specified values have a default value) and why you have to use `NA_real_` or `NA_character_` and the like instead of `NA`.

Reply:
Alright I think that did it! Thank you for the lesson @jsonbecker! Need to step away for a bit but I'll come back and if someone hasn't merged it by then I'll go ahead and take care of it later.